src/pybind/mgr/dashboard/controllers/__init__.py

```diff
@@ -662,14 +667,23 @@ def inner(*args, **kwargs):
             params = get_request_body_params(cherrypy.request)
             kwargs.update(params)
 
-            ret = func(*args, **kwargs)
+            req_version = '15'
+            if 'vnd.ceph.api.v' in cherrypy.request.headers.get('Content-Type'):
+                req_version = re.match(r"\d\d", cherrypy.request.headers['Content-Type'])
+                if req_version not in ['15']:
+                    req_version = '15'
```

Comment:
I'm curious, does this work with the current backend? Supposedly, Cherrypy json will check the Accept and Content-Type and throw a 415 if anything except `'application/json'` and `'text/javascript'` is received (vs. `application/vnd.ceph.api.vX.Y+json`).

Right now we are using CherryPy built-in handlers [`json_in`](https://github.com/cherrypy/cherrypy/blob/master/cherrypy/lib/jsontools.py#L16-L55) ~and [`json_out`](https://github.com/cherrypy/cherrypy/blob/master/cherrypy/lib/jsontools.py#L63-L89)~ (but we should) for processing json input/output, also including `Content-Type` negotiation.

One possibility could be to add a custom JSON handler before the Cherrypy's (and that'd be anything with [priority below 30](https://github.com/cherrypy/cherrypy/blob/2fc78dc9af3e556829ce69d3e1311e5629200882/cherrypy/_cptools.py#L496-L497)), and basically extract the version and overwrite the Accept/Content-Type with the standard `application/json`.

Other approach could be to simply copy and rewrite the CherryPy JSON tools to properly deal with `application/.*+json` Content-types.